### PR TITLE
[Merged by Bors] - Don't limit get hash requests for fetching ATX deps

### PR DESCRIPTION
**Pull request by poszu**
## Motivation
The original limiting of ATX fetching didn't consider the recursiveness of fetching an ATX (fetching an ATX can trigger fetching its dependencies). The simultaneous fetching of many ATXs could stall, waiting for a slot to fetch dependencies, without a way to unblock and starve.

## Changes
Don't limit ATX fetches when fetching dependencies of a processed ATX. Huge batch fetches of ATXs from processing an active set or epoch info are limited.

## Test Plan
I ran against the mainnet and didn't observe fetch stalls.

### CODE REVIEW

File: system/fetcher.go [hunk 1]
```diff
@@ -26,9 +26,21 @@ type BlockFetcher interface {
...
+	return func(opts *GetAtxOpts) {
+		opts.LimitingOff = true
+	}
+}
+
 // AtxFetcher defines an interface for fetching ATXs from remote peers.
 type AtxFetcher interface {
-	GetAtxs(context.Context, []types.ATXID) error
+	GetAtxs(context.Context, []types.ATXID, ...GetAtxOpt) error
 }
```
Comment (fasmat) [inline system/fetcher.go:44]: NIT: I feel like these options and the interface don't belong here; the options should be where the implementation is (`fetch` package) and the interface probably where it is used (`atxsync` and `activation` package?)

**Review by fasmat — APPROVED** [review]

**Comment by poszu** [comment]
bors merge

**Comment by poszu** [comment]
bors cancel

**Comment by poszu** [comment]
bors merge

**Review by pigmej — APPROVED** [review]

**Pull request closed without merging**